src/components/file-browser/DirectoryTree.ts

```diff
@@ -81,6 +88,14 @@ export const Subdirectories = (props: SubdirectoriesProps) => {
     }
   }, [directoryLabel, status])
 
+  useEffect(() => {
+    return reloadRequests.subscribe((pathToReload: string) => {
+      if (pathToReload === path) {
```

Comment:
That's correct. It's so that when you delete the current folder from the FilesInDirectory, its parent folder in the DirectoryTree reloads so that the deleted folder no longer appears there.